src/utils/ObjectUtils.ts

```diff
@@ -50,15 +50,20 @@ function isImmutable(obj: any): boolean {
   );
 }
 
-function _compareArrays(arr1: unknown[], arr2: unknown[]): boolean {
+function _compareArrays(arr1?: unknown[], arr2?: unknown[]): boolean {
+  if (!arr1 || !arr2) return false;
   if (arr1.length !== arr2.length) {
     return false;
   }
   if (!arr1.length) {
     return true;
   }
   for (const idx in arr1) {
-    if (arr1[idx] !== arr2[idx]) {
+    // this makes it _technically_ not a shallow equal for array values,
+    // but for stores that return arrays of objects, the object identities
+    // will likely not be equal so we should actually inspect the objects
+    // to see if they're equal
+    if (!shallowEqual(arr1[idx], arr2[idx])) {
```

Comment:
Should we still worry if for whatever reason you had an array and a truth value here for `arr2`?

Reply:
the early-exit conditions in `shallowEqual` should keep us safe there, they'd presumably fail the `typeof` check that we run https://github.com/HubSpot/general-store/pull/91/files#diff-e29b6fdae31394e0081186a94657c1d5R85-R97